File: c program/binarysearch.c
#include<stdio.h>
int main()
{
    int n , i;
    n = 65000;
    int arr[n];
   // printf("ENTER THE NUMBERS OF ELEMENT IN AN ARRAY:");
    //scanf("%d",&n);
   // for ( i=0;i<n;i++)
   // {
//scanf("%d",&arr[i]);
    //}
    for(i=0;i<n;i++)
    {
        arr[i] = i;
    }



    int number;
while (1)
{
    printf("ENTER THE NUMBER TO BE SEARCHED : ");
    scanf("%d",&number);
    int max ,min ,middle ;
    max = n-1 ;
    min =0;
    while ( min <= max)
    {
        middle = (max + min)/2 ;
        if (arr[middle] == number )
        {
            printf("%d is present at position %d", number ,middle);
            break;
        }
        else if (arr[middle] > number)
        {
            max = middle - 1;
        }
        else if(arr[middle] < number)
        {
            min  = middle +1 ;
        }
    }
    if (min > max)
        printf("THE NUMBER DOES NOT EXISTS ");
}
}

File: python-program/binarysearch.py
list2 =[1,2,5,7,9,11,25,67,77,89,94,101]
num = int(input("Enter the number to be searched:"))
n = len(list2)
min =0 
max =n-1
for i in range(n):
    avg = (min + max)//2
    if num == list2[max] :
        print("Number found at : ",max)
        break
    elif num == list2[min] :
        print("Number found ",min)
        break
    elif num == list2[avg]:
        print("number found ",avg)
        break
    elif num > list2[avg] :
          min = avg
       
    elif num <list2[avg] :
        max = avg
else :
    print("Nmuber not found ")

